src/state.rs

```diff
@@ -92,11 +92,16 @@ impl State {
         let encryption_key = self
             .get_encryption_key_from_state()
             .map_err(|_| iso7816::Status::SecurityStatusNotSatisfied)?;
-        let data = EncryptedDataContainer::from_obj(trussed, obj, None, encryption_key)
-            .map_err(|_| Status::UnspecifiedPersistentExecutionError)?;
-        let data_serialized: Message = data
-            .try_into()
-            .map_err(|_| Status::UnspecifiedPersistentExecutionError)?;
+        let data = EncryptedDataContainer::from_obj(trussed, obj, None, encryption_key).map_err(
+            |_err| {
+                error!("error serializing: {:?}", _err);
+                Status::UnspecifiedPersistentExecutionError
+            },
+        )?;
+        let data_serialized: Message = data.try_into().map_err(|_err| {
+            error!("error transforming: {:?}", _err);
+            Status::UnspecifiedPersistentExecutionError
+        })?;
```

Comment:
I think it would be more correct to call them `error encrypting object` and `error serializing container`

In the end it doesn't matter much since someone seeing these logs will likely only use them to find the line in the code.

Reply:
These are iso7816 codes, but yes - it should use its own error codes first, and translations second. Would that not make it harder to debug though?

Reply:
I'm talking about the log messages